chrome/content/zotero/xpcom/zotero.js

```diff
@@ -381,7 +381,26 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 					return false;
 				}
 			}
-			
+			if(Zotero.File.isDropboxDirectory(dataDir.path) && Zotero.Prefs.get('warnDropboxDataDir')){
```

Comment:
Yes, that makes sense for startup. Didn't look at this carefully enough. I'd rename it to "checkDataDirLocationOnStartup" (for clarity and because there might be other services to check for in the future) and move it to be the first check in the `if` statement.